support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/DelegatedClientAuthenticationAction.java

```diff
@@ -146,8 +150,13 @@ protected Event doExecute(final RequestContext context) throws Exception {
                 final ClientCredential clientCredential = new ClientCredential(credentials);
                 final AuthenticationResult authenticationResult =
                         this.authenticationSystemSupport.handleAndFinalizeSingleAuthenticationTransaction(service, clientCredential);
-                final TicketGrantingTicket tgt = this.centralAuthenticationService.createTicketGrantingTicket(authenticationResult);
-                WebUtils.putTicketGrantingTicketInScopes(context, tgt);
+                try {
+                    final TicketGrantingTicket tgt = this.centralAuthenticationService.createTicketGrantingTicket(authenticationResult);
+                    WebUtils.putTicketGrantingTicketInScopes(context, tgt);
+                } catch (final PrincipalException e) {
+                    LOGGER.warn("Principal attributes have not been validated on service [{}]", e.getMessage());
```

Comment:
Possibly should change to:

```
LOGGER.warn(e.getMessage(), e);
```

The error message that is indicated above does not make that much sense. Principal attributes cannot be validated in any way by CAS.